### DISCUSSION TITLE
Compensation has been removed from railsdevs

### DISCUSSION THREAD

**Discussion by joemasilotti** [joemasilotti/railsdevs.com#230] [Announcements]
I've removed all compensation fields, sorting, and display from railsdevs.
 
Yesterday morning I launched #203 which added filters to `/developers`. After sleeping on it, I realized I don't agree with the message that filtering by “budget” conveys. **Filtering developers by their hourly rates won’t connect the best developer for the job to the business.** And for a site whose top priority is to empower the independent developer, well, this feels backwards.
 
From my experience as a freelancer, **compensation should be discussed only after the details of a project or job are known**. When developers share their rate before then it limits their ability to get adequately compensated for their skills. It also encourages businesses to find the least expensive work which usually doesn’t line up with the quality they expect.
 
Instead, businesses should find the best fit candidate from developer bios. Then, after opening a real dialog with them, can they discuss what they need help with and their budget. **This enables developers to charge based on the complexity, scope, difficulty, and value of the work** – not just hours in and hours out.
 
Frankly, a business saying they have a budget for someone who charges $X per hour is misguided. How do you know if the developer can get it done in 1 hour or 50? Your budget is what you can spend (each month). It shouldn't matter how the developer breaks that down or what they charge per hour for the work if they get it done within that budget.
 
---
 
I'd love for this to be an open discussion. How do you, as a developer, feel about being asked for a rate before getting project details?

**Comment by Odaeus** [comment]
Hey Joe, I just followed the link here from the profile editing page. It's disappointing to see this removed as I believe in open pricing models and think it's better for all when pricing is more transparent. I have been a freelancer since 2006 and charge a flat rate per hour. I am happy being compensated on a time basis as it's fair and by far the most practical option. All of those aspects you mention "complexity, scope, difficulty, and value" are difficult to measure and harder to estimate. There's no way to accurately estimate the cost of a particular project without already investing significant time into determining these things, which is work that should be paid for! Even if you do this work, there are no guarantees about being able to deliver within a particular budget. Fixed budgets are more likely to lead to a freelancer being under-compensated as they expend more time trying to meet deliverables set by inaccurate estimations.

I've written my rate on my profile text, which is a perfectly fine workaround! 😄 

Thank you for making Rails Devs!

**Reply by joemasilotti** [reply]
Thanks for the feedback, I appreciate it. And kudos to you for getting a fair, flat rate for every project that you do. (And the nice workaround on railsdevs!)

In my experience, certain skills are more sought after or more valuable to certain companies than others. If I were to charge the same rate for everything I do then some clients would be overpaying and others would be underpaying. By removing a fixed rate from the site the developer and business are free to discuss what works best for the project, scope of work, and budget.

> Tangentially, hourly rates can limit your income as a freelancer to the amount of work you put in. Why should you get paid less the tenth time you've done something just because you can do it more quickly than before? Shouldn't you get paid _more_ because you now have unique, real-world experience from the other 9 times you've done it?

I agree that it's hard to estimate anything! Knowing what to charge is a skill on its own. Any project I work on will have a long list of very clear assumptions and expectations. If any of those are broken then both parties understand a fee adjustment might be needed.

I removed rate to protect developers from businesses trying to find the least expensive work. If you're confident in charging a flat rate then by all means keep it on your profile.
